core-libs/setup/ssr/error-handling/server-error-response-transformers/server-error-response-transformers.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { InjectionToken } from '@angular/core';
+import { Applicable } from '@spartacus/core';
+import { CxServerError } from '../server-errors';
+
+/**
+ * A transformer responsible for transforming an error into a {@link CxServerError}.
+ * The transformer is applicable when the error matches the transformer's criteria.
+ * Spartacus provides two default transformers:
+ * - {@link CmsPageNotFoundServerErrorResponseTransformer}
+ * - {@link UnknownServerErrorResponseTransformer}
+ */
+export interface ServerErrorResponseTransformer extends Applicable {
+  transform(error: unknown): CxServerError;
+}
+
+/**
+ * Injection token for server error response transformers.
+ */
+export const SERVER_ERROR_RESPONSE_TRANSFORMERS = new InjectionToken<
+  ServerErrorResponseTransformer[]
+>('SERVER_ERROR_RESPONSE_TRANSFORMERS');
```

Comment:
We proposed `Transformer` from top of head, but knew it's not the best name for it.
Reminder to decide on naming of it.
- ServerErrorResponseTransformer .transform(unknown): CxServerError
- ServerErrorResponseMapper .map(unknown): CxServerError
- ServerErrorResponseStrategy .handle(unknown): CxServerError
- ServerErrorResponseHandler .handle(unknown): CxServerError

Btw. there is an inconsistency. Sometimes we say ServerErrorResponse, and other times ServerError, which confused me a bit. I guess it wasn't deliberate?

Reply:
As discussed, this abstraction layer is, in fact, the factory pattern. 
I'll change `SERVER_ERROR_RESPONSE_TRANSFORMERS` into `SERVER_ERROR_RESPONSE_FACTORY` and adjust the related code.